### PR TITLE
make install: install runc and containerd

### PR DESCRIPTION
**Pull request by AkihiroSuda**
**\- What I did**
Install runc and containerd in `make install`

**\- How I did it**
Please read the code.

**\- How to verify it**
`make install && ls -l /usr/local/bin/docker*`

Signed-off-by: Akihiro Suda suda.akihiro@lab.ntt.co.jp

### CODE REVIEW

File: hack/make/.binary-setup [hunk 1]
```diff
@@ -2,4 +2,8 @@
 
 DOCKER_CLIENT_BINARY_NAME='docker'
 DOCKER_DAEMON_BINARY_NAME='dockerd'
+DOCKER_RUNC_BINARY_NAME='docker-runc'
+DOCKER_CONTAINERD_BINARY_NAME='docker-containerd'
+DOCKER_CONTAINERD_CTR_BINARY_NAME='docker-containerd-ctr'
+DOCKER_CONTAINERD_SHIM_BINARY_NAME='docker-containerd-shim'
```
Comment (jfrazelle) [inline hack/make/.binary-setup]: maybe we could do a DOCKER_BIN_PREFIX=docker- or idk, i guess thats grosser, this is fine

**Comment by cpuguy83** [comment]
The reason I didn't do these is because `make binary` already installs them.

**Comment by thaJeztah** [comment]
> The reason I didn't do these is because make binary already installs them.

ah! I was already scratching my head why it worked without this change

@AkihiroSuda was there a reason for adding this? Did you run into issues?

**Comment by AkihiroSuda** [comment]
Sorry, I could not get it ..

In my understanding `make binary` does not install the binaries to `${DOCKER_MAKE_INSTALL_PREFIX}/bin`.

`make install` does that, but it does not install `runc` and `containerd`
```
$ DOCKER_MAKE_INSTALL_PREFIX=/tmp/a make install
Cannot connect to the Docker daemon. Is the docker daemon running on this host?
Cannot connect to the Docker daemon. Is the docker daemon running on this host?
Cannot connect to the Docker daemon. Is the docker daemon running on this host?
Cannot connect to the Docker daemon. Is the docker daemon running on this host?
KEEPBUNDLE=1 hack/make.sh install-binary
# WARNING! I don't seem to be running in a Docker container.
# The result of this command might be an incorrect build, and will not be
# officially supported.
#
# Try this instead: make all
#

---> Making bundle: install-binary (in bundles/1.12.0-dev/install-binary)
Installing docker to /tmp/a/bin/
Installing dockerd to /tmp/a/bin/
Installing docker-proxy to /tmp/a/bin/

$ ls /tmp/a/bin/
docker  dockerd  docker-proxy
```

**Comment by godmac** [comment]
how could i unsubscribe this mailist

2016年7月8日星期五，Akihiro Suda <notifications@github.com> 写道：

**Comment by thaJeztah** [comment]
@godmac you can unsubscribe from the docker/docker repository, or just this thread (using the unsubscribe button)

**Comment by cpuguy83** [comment]
@AkihiroSuda It's handled here: https://github.com/docker/docker/blob/master/hack/make/binary-daemon#L15

Which is defined here: https://github.com/docker/docker/blob/master/hack/make.sh#L319

**Comment by AkihiroSuda** [comment]
@cpuguy83 Thank you, but I think we are still talking about different things..

The `copy_containerd` you mentioned usually works _INSIDE_ of the `docker-dev` container, and copies `/usr/local/bin/docker-{containerd, containerd-shim, containerd-ctr, runc}` _INSIDE_ of the container to the `bundles/latest/binary-daemon` directory.

On the other hand, my modified `make install` works _OUTSIDE_ of the `docker-dev` container, and copies `bundles/latest/binary-daemon/docker-{containerd, containerd-shim, containerd-ctr, runc}` to `${DOCKER_MAKE_INSTALL_PREFIX}/bin` _OUTSIDE_ of the container. (I mean, the `/usr/local/bin` directory (by default) on the "baremetal" host)

Hope it makes sense.. :smile:

**Comment by cpuguy83** [comment]
@AkihiroSuda Ah silly me... you are right.
Though now when using the script in a dev container we will re-copy the files back into usr/local/bin.

**Comment by cpuguy83** [comment]
Otherwise, LGTM. I can live with the double copy.

**Comment by LK4D4** [comment]
LGTM

**Pull request merged**